src/EventListener/CreateSchemaListener.php

```diff
@@ -74,9 +83,33 @@ public function postGenerateSchemaTable(GenerateSchemaTableEventArgs $eventArgs)
         );
 
         foreach ($entityTable->getColumns() as $column) {
+            $columnTypeName = $column->getType()->getName();
+            $columnArrayOptions = $column->toArray();
+
+            //ignore specific fields for subclasses in-case of using discriminator column
+            foreach ($cm->subClasses as $subClass) {
+                if ($cm->hasField($column->getName()) || $cm->hasAssociation($column->getName())) {
+                    if ($this->config->isEntityIgnoredProperty($subClass, $cm->getFieldForColumn($column->getName()))) {
+                        continue 2;
+                    }
+                }
+            }
+
+            //ignore specific fields for table
+            if (empty($cm->discriminatorColumn) && $this->config->isEntityIgnoredProperty($cm->getName(), $cm->getFieldForColumn($column->getName()))) {
+                continue;
+            }
+
+            // change Enum type to String
+            $sqlString = $column->getType()->getSQLDeclaration($columnArrayOptions, $this->em->getConnection()->getDatabasePlatform());
+            if ($this->config->convertEnumToString() && false !== strpos($sqlString, 'ENUM')) {
```

Comment:
What are the possible strings containing "ENUM" as substring?

Reply:
`ENUM('XXX', 'YYY', 'ZZZZ')`

Reply:
IMO, then we could use `0 === strpos($sqlString, 'ENUM(')` in order to improve the search, since we already know that the string must start with the required substring.